docs/source/mods/metromap.rst

```diff
@@ -0,0 +1,511 @@
+  the directions 0, 1, and 2 are allowed for an edge :math:`(u,v)`, i.e.,
+  :math:`d(u,v,3)=d(u,v,4)=d(u,v,5)=d(u,v,6)=d(u,v,7)=0`, we are in the positive
+  part of the coordinate system and the following constraints must hold
+
+  .. math::
+    \begin{alignat}{2}
+      x(v) & \geq x(u)\\
+      y(v) & \geq y(u) \\
+      x(v) + y(v) & > x(u) + y(u)
+    \end{alignat}
+
+  Similar constraints result when considering all other combinations of 3
+  adjacent directions. We add these conditions as big-M constraints to the model
+  and use an epsilon of 1e-6 (=feasibility tolerance) to handle the strict
+  inequality.
+
+  For the bend variables, the following constraint can be added
+
+  .. math::
+      \sum_{(v,w)\in E, w\neq u} b(u,v,w,0) \leq 1 \quad \forall (u,v) \in E
+
+  The constraint states that edge :math:`(u,v)` can have a bend of 0 with at
+  most one of its adjacent edges.
+
+Code and Inputs
+---------------
+
+For this OptiMod, the ``networkx`` package is needed. The input is a
+``networkx`` graph for the network information and a ``pd.DataFrame`` for the line
+information. The output is a ``networkx`` graph with node property ``pos_oct``
+representing the x-and y-coordinates for the octilinear representation and a
+Python dictionary for the direction of each edge.
+
+An example of the inputs with the respective requirements is shown below.
+
+
+.. doctest:: load_graph
+    :options: +NORMALIZE_WHITESPACE
+
+    >>> import networkx as nx
+    >>> from gurobi_optimods import datasets
+    >>> graph, linepath_data = datasets.load_sberlin_graph_data()
+    >>> print(graph)
+      Graph with 167 nodes and 175 edges
+    >>> pos_orig = nx.get_node_attributes(graph, 'pos')
+    >>> len(pos_orig)
+      167
+    >>> linepath_data.head(4)
+      linename	edge_source	edge_target
+      0	S1	100	78
+      1	S1	78	23
+      2	S1	23	20
+      3	S1	20	60
+
+For the example, we used data from the S-Bahn Berlin network. The graph includes
+the node attribute ``pos`` that contains a tuple of x- and y-coordinates for
+the original position. The linepath_data must be consistent with
+the graph. For example, ``edge_source``, ``edge_target`` in the
+``linepath_data`` must correspond to the node names in the graph.
+
+
+Solution
+--------
+
+The output of the optimization is the following:
+
+- A graph with the node attribute ``pos_oct`` containing the x- and y-coordinates
+  for the octilinear representation.
+- A Python dictionary providing the direction (0 to 7) for each edge. This
+  information is needed for the plotting function provided in the OptiMod
+
+The OptiMod can be run as follows:
+
+.. testcode:: solve
+
+    from gurobi_optimods import datasets
+    from gurobi_optimods.metromap import metromap
+    graph, linepath_data = datasets.load_sberlin_graph_data()
+    graph_out, edge_directions = metromap(graph, linepath_data, include_planarity=False, verbose=False, time_limit=2)
```

Comment:
Also, can you format this line so that is is within 80 characters (have to scroll sideways to see). You can try to format all these bits of code using black.